### ISSUE TITLE
After a fresh clone, yarn install never finishes building fresh packages (Win 7 x64)

### ISSUE THREAD

**Issue by Neonit** [microsoft/vscode#77717]
- VSCode Version: Repo HEAD (4fc00c05ffcf6274cdea6eadc5df13ffbceebf69)  (Sat Jul 20 05:15:08 2019 +0200)
- OS Version: Windows 7 Professional SP1 (64 Bit)

Steps to Reproduce:

1. `git clone https://github.com/microsoft/vscode.git`
2. `cd vscode`
3. `yarn install`

- Node 10.15.3
- Yarn 1.17.3

Output:
```
yarn install v1.17.3
info No lockfile found.
$ node build/npm/preinstall.js
[1/4] Resolving packages...
warning coveralls > request > hawk@3.1.3: This module moved to @hapi/hawk. Pleas
e make sure to switch over as this distribution is no longer supported and may c
ontain bugs and critical security issues.
warning coveralls > request > hawk > hoek@2.16.3: This version has been deprecat
ed in accordance with the hapi support policy (hapi.im/support). Please upgrade
to the latest version to get the best features, bug fixes, and security patches.
 If you are unable to upgrade at this time, paid support is available for older
versions (hapi.im/commercial).
warning coveralls > request > hawk > cryptiles@2.0.5: This version has been depr
ecated in accordance with the hapi support policy (hapi.im/support). Please upgr
ade to the latest version to get the best features, bug fixes, and security patc
hes. If you are unable to upgrade at this time, paid support is available for ol
der versions (hapi.im/commercial).
warning coveralls > request > hawk > sntp@1.0.9: This module moved to @hapi/sntp
. Please make sure to switch over as this distribution is no longer supported an
d may contain bugs and critical security issues.
warning coveralls > request > hawk > sntp > hoek@2.16.3: This version has been d
eprecated in accordance with the hapi support policy (hapi.im/support). Please u
pgrade to the latest version to get the best features, bug fixes, and security p
atches. If you are unable to upgrade at this time, paid support is available for
 older versions (hapi.im/commercial).
warning coveralls > request > hawk > boom@2.10.1: This version has been deprecat
ed in accordance with the hapi support policy (hapi.im/support). Please upgrade
to the latest version to get the best features, bug fixes, and security patches.
 If you are unable to upgrade at this time, paid support is available for older
versions (hapi.im/commercial).
warning coveralls > request > hawk > cryptiles > boom@2.10.1: This version has b
een deprecated in accordance with the hapi support policy (hapi.im/support). Ple
ase upgrade to the latest version to get the best features, bug fixes, and secur
ity patches. If you are unable to upgrade at this time, paid support is availabl
e for older versions (hapi.im/commercial).
warning coveralls > request > hawk > boom > hoek@2.16.3: This version has been d
eprecated in accordance with the hapi support policy (hapi.im/support). Please u
pgrade to the latest version to get the best features, bug fixes, and security p
atches. If you are unable to upgrade at this time, paid support is available for
 older versions (hapi.im/commercial).
warning cson-parser > coffee-script@1.12.7: CoffeeScript on NPM has moved to "co
ffeescript" (no hyphen)
warning electron-mksnapshot > electron-download > nugget > progress-stream > thr
ough2 > xtend > object-keys@0.4.0:
warning eslint > file-entry-cache > flat-cache > circular-json@0.3.3: CircularJS
ON is in maintenance only, flatted is its successor.
warning gulp-cssnano > cssnano > autoprefixer > browserslist@1.7.7: Browserslist
 2 could fail on reading Browserslist >3.0 config used in other tools.
warning gulp-cssnano > cssnano > postcss-merge-rules > browserslist@1.7.7: Brows
erslist 2 could fail on reading Browserslist >3.0 config used in other tools.
warning gulp-cssnano > cssnano > postcss-merge-rules > caniuse-api > browserslis
t@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in o
ther tools.
warning gulp-cssnano > cssnano > postcss-merge-rules > postcss-selector-parser >
 flatten@1.0.2: I wrote this module a very long time ago; you should use somethi
ng else.
warning mocha > glob > minimatch@0.3.0: Please update to minimatch 3.0.2 or high
er to avoid a RegExp DoS issue
warning mocha > to-iso-string@0.0.2: to-iso-string has been deprecated, use @seg
ment/to-iso-string instead.
warning mocha > jade@0.26.3: Jade has been renamed to pug, please install the la
test version of pug instead of jade
warning sinon > formatio@1.1.1: This package is unmaintained. Use @sinonjs/forma
tio instead
warning sinon > samsam@1.1.2: This package has been deprecated in favour of @sin
onjs/samsam
warning sinon > formatio > samsam@1.1.3: This package has been deprecated in fav
our of @sinonjs/samsam
[2/4] Fetching packages...
info vscode-fsevents@1.2.12: The platform "win32" is incompatible with this modu
le.
info "vscode-fsevents@1.2.12" is an optional dependency and failed compatibility
 check. Excluding it from installation.
info fsevents@1.2.9: The platform "win32" is incompatible with this module.
info "fsevents@1.2.9" is an optional dependency and failed compatibility check.
Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[14/17] ? vscode-windows-registry
[17/17] ? windows-process-tree
[16/17] ? windows-mutex
[8/17] ? vscode-ripgrep
[15/17] ? windows-foreground-love
```

With this it just hangs. I left it running for half an hour and it didn't finish. While it had been working before, I could notice the sounds of my drives and my CPU cooler, but once it got to this state it appeared drives and CPU went to sleep.

Tested in the GIT Bash and also the default Windows CMD. GIT Bash displays much less information.

**Comment by joaomoreno** [comment]
Did you go through https://github.com/microsoft/vscode/wiki/How-to-Contribute ?

**Comment by Neonit** [comment]
I just followed https://github.com/Microsoft/vscode/wiki/How-to-Contribute#build-and-run. I just tried again with all the instruction steps and ran into the issues 2.1, 2.2 and 2.3 described on https://github.com/joquijada/windows-node-js-node-gyp. I followed the fixes described there and ended up with a completely different error (than 2.4).

```shell
/usr/bin/bash: /cygdrive/c/msys64/usr/bin/cmd: No such file or directory
gyp: Call to 'cmd /c "cd shared && GetCommitHash.bat"' returned exit status 127
while in deps/winpty/src/winpty.gyp. while loading dependencies of binding.gyp w
hile trying to load binding.gyp
```

I don't know why it looks for a `cmd` there. It's probably, because my sytem is messed up, because I have so much stuff installed (Cygwin, MinGW, MSYS, GIT Bash).
